interactions/utils/dict_caches.py

```diff
@@ -45,3 +47,12 @@ def __setitem__(self, key: _KT, value: _VT):
         # Prevent buildup over time
         while len(self) > self._max_items:
             del self[next(iter(self))]
+
+    def pop(self, key: _KT, default: _VT = MISSING) -> _VT:
```

Comment:
I can see cases where `MISSING` is a default that's passed, but it would unexpectedly raise a KeyError. Can you change it to what OrderedDict does, with a unique object?

Reply:
(Something similar to [what `collections` does](https://github.com/python/cpython/blob/main/Lib/collections/__init__.py#L231-L233))
![image](https://user-images.githubusercontent.com/84055084/202831245-15b4bac9-90ba-4816-8afd-72935e2991a6.png)

Reply:
Sure I'll do it, although I'd like to know when MISSING is passed in?

Reply:
Why do you need to know when MISSING is passed?